requirements/constraints.txt

```diff
@@ -48,3 +48,40 @@ python3-saml==1.5.0
 
 # version 1.3.0 breaks make upgrade, as it requires py3.5 or above
 scipy<=1.2.1
+
+# Leaving this unpinned breaks make upgrade, since several packages require <1.0.0, but pip-compile
+# will find and select 1.0.0 for paver.txt and terminate unsuccessfully.
+edx-opaque-keys==0.4.4
+
+# Unless we constrain pytest to <4.6.0, it may break all tests inside of
+# xmodule.tests.test_resource_templates.ResourceTemplatesTests for reasons that
+# I cannot explain.  I've copied the error below:
+#
+#
+#   self = <xmodule.tests.test_resource_templates.ResourceTemplatesTests testMethod=test_get_template>
+#
+#       def test_get_template(self):
+#           self.assertEqual(
+#   >           TestClass.get_template('latex_html.yaml')['template_id'],
+#               'latex_html.yaml')
+#
+#   common/lib/xmodule/xmodule/tests/test_resource_templates.py:28:
+#   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
+#   common/lib/xmodule/xmodule/x_module.py:1094: in get_template
+#       if resource_exists(pkg, path):
+#   ../venvs/edxapp/local/lib/python2.7/site-packages/pkg_resources/__init__.py:1134: in resource_exists
+#       return get_provider(package_or_requirement).has_resource(resource_name)
+#   ../venvs/edxapp/local/lib/python2.7/site-packages/pkg_resources/__init__.py:1404: in has_resource
+#       return self._has(self._fn(self.module_path, resource_name))
+#   _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
+#
+#   self = <pkg_resources.NullProvider object at 0x7f4164c4c510>, path = '/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/tests/templates/test/latex_html.yaml'
+#
+#       def _has(self, path):
+#           raise NotImplementedError(
+#   >           "Can't perform this operation for unregistered loader type"
+#           )
+#   E       NotImplementedError: Can't perform this operation for unregistered loader type
```

Comment:
Yay verbosity!

Reply:
Maybe put the stack trace in a JIRA ticket and just link to the ticket here?

Reply:
Added this info to TE-2391